share/functions/fish_prompt.fish

```diff
@@ -23,7 +23,26 @@ function fish_prompt --description 'Write out the prompt'
     end
 
     # Write pipestatus
-    set -l prompt_status (__fish_print_pipestatus $last_status " [" "]" "|" (set_color $fish_color_status) (set_color --bold $fish_color_status) $last_pipestatus)
+    set -l bold_flag '--bold'
+    if $__fish_prompt_status_is_holdover
+        set bold_flag
+    end
+    set -l prompt_status (__fish_print_pipestatus $last_status " [" "]" "|" (set_color $fish_color_status) (set_color $bold_flag $fish_color_status) $last_pipestatus)
 
     echo -n -s (set_color $fish_color_user) "$USER" $normal @ (set_color $color_host) (prompt_hostname) $normal ' ' (set_color $color_cwd) (prompt_pwd) $normal (fish_vcs_prompt) $normal $prompt_status $suffix " "
 end
+
+set __fish_prompt_status_generation $status_generation
+set __fish_prompt_status_is_holdover false
+
+function __fish_prompt_save_generation --on-event fish_preexec
+    set __fish_prompt_status_generation $status_generation
+end
+
+function __fish_prompt_dim_holdover_status --on-event fish_postexec
+    if test $__fish_prompt_status_generation = $status_generation
```

Comment:
nit: (this hook is no longer necessary, we can do the comparison in `fish_prompt`)

Reply:
Good call. Done. I wasn't too happy with all the underscores anyway